### ISSUE TITLE
Loss returns nan on GPU 

### ISSUE THREAD

**Issue by Belencostanza** [keras-team/keras#15995]
Hi,

I have an issue running the example network "Simple MNIST convnet" in the keras documentation. My losses are nan when I'm running on GPU nodes in a cluster while in CPU is working fine, I'm using a conda environment with tensorflow 2.4.1 and tensorflow-gpu 2.4.1 .... the keras backend is tensorflow. 

I figured out that the problem appears with losses that make the operation tf.reduce_mean. For example, with sparse_categorical_crossentropy and reduction='none' the losses are fine. 

Thanks!

**Comment by jvishnuvardhan** [comment]
@Belencostanza Can you please share a simple standalone code to reproduce the issue?  Also, mention which distribution strategies you are using. Add as many details as follows. Thanks!

**Comment by Belencostanza** [comment]
Here is the code extracted from the keras documentation: https://keras.io/examples/vision/mnist_convnet/

**# Model / data parameters
num_classes = 10
input_shape = (28, 28, 1)

# the data, split between train and test sets
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

# Scale images to the [0, 1] range
x_train = x_train.astype("float32") / 255
x_test = x_test.astype("float32") / 255
# Make sure images have shape (28, 28, 1)
x_train = np.expand_dims(x_train, -1)
x_test = np.expand_dims(x_test, -1)
print("x_train shape:", x_train.shape)
print(x_train.shape[0], "train samples")
print(x_test.shape[0], "test samples")


# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)


model = keras.Sequential(
    [
        keras.Input(shape=input_shape),
        layers.Conv2D(32, kernel_size=(3, 3), activation="relu"),
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Flatten(),
        layers.Dropout(0.5),
        layers.Dense(num_classes, activation="softmax"),
    ]
)
batch_size = 128
epochs = 5
model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=0.1)**

then my conda list:
[list.txt](https://github.com/keras-team/keras/files/7987640/list.txt)

I only did
$ conda create --name gpu_prueba tensorflow-gpu

Important libraries: 
tensorflow-gpu 2.4.1
cudatoolkit 10.1.243
cudnn 7.6.5
python 3.9.7



I connect via ssh to a cluster with nodes NVIDIA A100 40GB GPUs

**Comment by Belencostanza** [comment]
I already resolved the problem when I installed specific versions with: 

$ conda create --name gpu-c11.2 -c conda-forge tensorflow-gpu matplotlib cudatoolkit=11.2 

cudatoolkit 11.2.2 he111cf0_10 conda-forge
python 3.9.10 h85951f9_2_cpython conda-forge
tensorflow 2.6.2 cuda112py39h23446aa_2 conda-forge
tensorflow-gpu 2.6.2 cuda112py39h0bbbad9_2 conda-forge
keras 2.6.0 pyhd8ed1ab_1 conda-forge
cudnn 8.2.1.32 h86fa8c9_0 conda-forge

Thanks a lot, I will close de issue

**Comment by MoniFarsang** [comment]
@Belencostanza, I have no idea how you figured out that this happens due to the tf.reduce_mean, but you saved my mental health. The issue exists not only on NVIDIA A100, but also on A40 and A6000. However, it doesn't occur on T4 or GeForce RTX 3050. Also, the nans appear sometimes at the very beginning of the training and sometimes only during the training. There are no error messages or warnings.

The package update indeed solves the issue.
